pkg/rotator/rotator.go

```diff
@@ -83,10 +85,11 @@ type WebhookInfo struct {
 
 func (w WebhookInfo) gvk() schema.GroupVersionKind {
 	t2g := map[WebhookType]schema.GroupVersionKind{
-		Validating:    {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "ValidatingWebhookConfiguration"},
-		Mutating:      {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "MutatingWebhookConfiguration"},
-		CRDConversion: {Group: "apiextensions.k8s.io", Version: "v1", Kind: "CustomResourceDefinition"},
-		APIService:    {Group: "apiregistration.k8s.io", Version: "v1", Kind: "APIService"},
+		Validating:           {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "ValidatingWebhookConfiguration"},
+		Mutating:             {Group: "admissionregistration.k8s.io", Version: "v1", Kind: "MutatingWebhookConfiguration"},
+		CRDConversion:        {Group: "apiextensions.k8s.io", Version: "v1", Kind: "CustomResourceDefinition"},
+		APIService:           {Group: "apiregistration.k8s.io", Version: "v1", Kind: "APIService"},
+		ExternalDataProvider: {Group: "externaldata.gatekeeper.sh", Version: "v1beta1", Kind: "Provider"},
```

Comment:
I wonder at what point it makes sense to replace the object-specific code with more generic code (e.g. watch this Kind, make sure certs live at path X)

Reply:
That could make sense. @maxsmythe would you mind if the discussion/implementation is delegated to a separate issue?

Reply:
Not at all